test/quantization/core/test_quantized_tensor.py

```diff
@@ -783,6 +783,32 @@ def test_qtensor_fill(self):
             self.assertEqual(q_filled.q_scale(), scale)
             self.assertEqual(q_filled.q_zero_point(), zero_point)
 
+    @unittest.skipIf(not TEST_CUDA, "No gpu is available.")
+    def test_qtensor_index_select_cuda(self):
+        self._test_qtensor_index_select('cuda')
+
+    def test_qtensor_index_select_cpu(self):
+        self._test_qtensor_index_select('cpu')
+
+    def _test_qtensor_index_select(self, device):
+        for quant_type in [torch.quint8, torch.qint8]:
+            dims = 3
+            index = torch.randint(dims, [1]).item()
+            selected = torch.randperm(dims)[:2].to(device)
+            scale = 1
+            zp = 0
+
+            x = torch.randn([3] * dims, device=device) * 10
+            q = torch.quantize_per_tensor(x, scale, zp, quant_type)
+            qs = torch.index_select(q, index, selected)
+            for altered_index in range(2):
```

Comment:
should we just compare the following two path matches:
1. float tensor -> index_select -> quantize
2. float tensor -> quantize -> index_select

Reply:
oh yeah thats much better, fixed in newest version